aspnet/testing/unit-testing.rst

```diff
@@ -2,28 +2,131 @@ Unit Testing
 
 In this article:
-  - `Unit Testing`_
+  - `Getting Started with Testing`_
   - `Creating Test Projects`_
   - `Running Tests`_
 
+`Download sample from GitHub <https://github.com/aspnet/docs/tree/1.0.0-beta8/aspnet/testing/unit-testing/sample>`_. 
 
-`Download sample from GitHub <https://github.com/aspnet/docs/tree/master/aspnet/testing/unit-testing/sample>`_. 
+Getting Started with Testing
+----------------------------
+Having a suite of automated tests is one of the best ways to ensure a software application does what its authors intended it to do. There are many different kinds of tests for software applications, including :doc:`integration tests <integration-testing>`, web tests, load tests, and many others. At the lowest level are unit tests, which test individual software components or methods. Unit tests should only test code within the developer's control, and should not test infrastructure concerns, like databases, file systems, or network resources. Unit tests may be written using `Test Driven Development(TDD) <http://deviq.com/test-driven-development/>`_, or they can be added to existing code to confirm its correctness. In either case, they should be small, well-named, and fast, since ideally you will want to be able to run hundreds of them before pushing your changes into the project's shared code repository.
 
-Unit Testing
-------------
-About unit testing
+.. note:: A `unit test naming convention <http://ardalis.com/unit-test-naming-convention>`_ I've found to be very effective is to name the class and test such that it forms a sentence, e.g. ``PricingCalculatorShould.ReturnZeroGivenQuantityZero()``. This makes for very readable test results, and tends to be far more helpful than naming test classes "FooTests" or "BarTests".
+
+When writing unit tests, be careful you don't accidentally introduce dependencies on infrastructure. These tend to make tests slower and more brittle, and thus should be reserved for integration tests. You can avoid these hidden dependencies in your application code by following the `Explicit Dependencies Principle <http://deviq.com/explicit-dependencies-principle/>`_ and using :doc:`/fundamentals/dependency-injection` to request your dependencies from the framework. You can also keep your unit tests in a separate project from your integration tests, and ensure your unit test project doesn't have references to or dependencies on infrastructure packages.
 
 Creating Test Projects
 ----------------------
-Creating test projects using test runners.
+A test project is just a class library with references to a test runner and the project being tested (also referred to as the System Under Test or SUT). It's a good idea to organize your test projects in a separate folder from your SUT projects, and the recommended convention for DNX projects is something like this:
+
+.. code-block:: text
+
+	global.json
+	PrimeWeb.sln
+	src/
+	   PrimeWeb/
+	      project.json
+		  Startup.cs
```

Comment:
the spacing looks off here. Startup.cs and project.json should be on the same level.